tests/system/test_dbapi.py

```diff
@@ -59,779 +57,757 @@ def raw_database(shared_instance, database_operation_timeout, not_postgres):
-    """Check that DDL statement followed by non-DDL execute statement in
-    non autocommit mode results in successful DDL statement execution."""
+        # read the resulting data from the database
+        self._cursor.execute("SELECT * FROM DdlExecuteMany")
+        got_rows = self._cursor.fetchall()
 
-    conn = Connection(shared_instance, dbapi_database)
-    want_row = (
-        1,
-        "first-name",
-    )
-    cur = conn.cursor()
-    cur.execute(
-        """
-        CREATE TABLE DdlExecute (
-            SingerId     INT64 NOT NULL,
-            Name    STRING(1024),
-        ) PRIMARY KEY (SingerId)
-        """
-    )
-    table = dbapi_database.table("DdlExecute")
-    assert table.exists() is False
+        assert got_rows == [want_row]
 
-    cur.execute(
+    @pytest.mark.skipif(_helpers.USE_EMULATOR, reason="Emulator does not support json.")
+    def test_autocommit_with_json_data(self, dbapi_database):
         """
-        INSERT INTO DdlExecute (SingerId, Name)
-        VALUES (1, "first-name")
+        Check that DDLs in autocommit mode are immediately
+        executed for json fields.
         """
-    )
-    assert table.exists() is True
-    conn.commit()
-
-    # read the resulting data from the database
-    cur.execute("SELECT * FROM DdlExecute")
-    got_rows = cur.fetchall()
-
-    assert got_rows == [want_row]
-
-    cur.close()
-    conn.close()
-
-
-def test_ddl_executemany(shared_instance, dbapi_database):
-    """Check that DDL statement followed by non-DDL executemany statement in
-    non autocommit mode results in successful DDL statement execution."""
+        try:
+            self._conn.autocommit = True
+            self._cursor.execute(
+                """
+                CREATE TABLE JsonDetails (
+                    DataId     INT64 NOT NULL,
+                    Details    JSON,
+                ) PRIMARY KEY (DataId)
+            """
+            )
+
+            # Insert data to table
+            self._cursor.execute(
+                sql="INSERT INTO JsonDetails (DataId, Details) VALUES (%s, %s)",
+                args=(123, JsonObject({"name": "Jakob", "age": "26"})),
+            )
+
+            # Read back the data.
+            self._cursor.execute("""select * from JsonDetails;""")
+            got_rows = self._cursor.fetchall()
+
+            # Assert the response
+            assert len(got_rows) == 1
+            assert got_rows[0][0] == 123
+            assert got_rows[0][1] == {"age": "26", "name": "Jakob"}
+
+            # Drop the table
+            self._cursor.execute("DROP TABLE JsonDetails")
+            self._conn.commit()
+        finally:
+            # Delete table
+            table = dbapi_database.table("JsonDetails")
+            if table.exists():
+                op = dbapi_database.update_ddl(["DROP TABLE JsonDetails"])
+                op.result()
+
+    @pytest.mark.skipif(_helpers.USE_EMULATOR, reason="Emulator does not support json.")
```

Comment:
This and the following comments are on the existing code which is not updated in this PR, its just showing as a new created code because of refactoring. 
This doesn't mean that I meant to not fix them (probably not in this PR :) ) but just wanted to let you know incase you are thinking its the new test being written.
Let me get the answers to these as I am also not aware

Reply:
Checked with Knut that we support it now. Will enable it in a separate PR